scheduler/scheduler.go

```diff
@@ -281,13 +291,32 @@ func generateContainerHostIntents(d distro.Distro, newContainersNeeded int) ([]h
 	return containerHostIntents, nil
 }
 
+func getDockerOptionsFromProviderSettings(settings *map[string]interface{}) (*host.DockerOptions, error) {
+	dockerOptions := &host.DockerOptions{}
+	if settings != nil {
+		if err := mapstructure.Decode(settings, dockerOptions); err != nil {
+			return nil, errors.Wrap(err, "Error decoding params")
+		}
+	}
+	if dockerOptions.Image == "" {
+		return nil, errors.New("docker image cannot be empty")
+	}
+	return dockerOptions, nil
+}
+
 // generateParentHostOptions generates host options for a parent host
-func generateParentHostOptions(pool *evergreen.ContainerPool) cloud.HostOptions {
-	return cloud.HostOptions{
+func generateParentHostOptions(settings *map[string]interface{}, pool *evergreen.ContainerPool) (cloud.HostOptions, error) {
+	options := cloud.HostOptions{
 		HasContainers:         true,
 		UserName:              evergreen.User,
 		ContainerPoolSettings: pool,
 	}
+	opts, err := getDockerOptionsFromProviderSettings(settings)
+	if err != nil {
+		return options, errors.Wrap(err, "error getting docker options from settings")
```

Comment:
don't return an error and a non-nil options document, when possible.

Reply:
The main reason I did this is because in createParents I'm not convinced we should error if the parent doesn't have the correct docker options--is there a case this could happen? For example does a parent have an image? That's why I put a debug message but continued execution with whatever we got back. If a parent doesn't have an image than should that be an error? If so then an empty options makes sense (and I should move the image validation).

Reply:
I was definitely confused about what was going on here and this doesn't even need to happen.